modules/etcpak/SCsub

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+
+Import("env")
+Import("env_modules")
+
+env_etcpak = env_modules.Clone()
+
+# Thirdparty source files
+thirdparty_dir = "#thirdparty/etcpak/"
+thirdparty_sources = [
+    "Bitmap.cpp",
+    "BitmapDownsampled.cpp",
+    "BlockData.cpp",
+    "ColorSpace.cpp",
+    "DataProvider.cpp",
+    "Debug.cpp",
+    "Dither.cpp",
+    "Error.cpp",
+    "mmap.cpp",
+    "ProcessDxtc.cpp",
+    "ProcessRGB.cpp",
+    "System.cpp",
+    "Tables.cpp",
+    "TaskDispatch.cpp",
+    "Timing.cpp",
+    "lz4/lz4.c",
+]
+thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
+
+env_etcpak.Append(CPPPATH=[thirdparty_dir, "#thirdparty/"])
+env_etcpak.Append(CPPDEFINES=["NO_GZIP"])
+env_etcpak.Append(CPPDEFINES=["PNG_INTEL_SSE"])
+if env["platform"] != "osx":
+    env_etcpak.Append(CPPDEFINES=["__SSE4_1__"])
+    env_etcpak.Append(CPPDEFINES=["__AVX__"])
+    env_etcpak.Append(CPPDEFINES=["__AVX2__"])
```

Comment:
The same CPU compatibility concerns apply here. AVX and AVX2 are unavailable on many low-end Intel CPUs due to market segmentation, even recent models.

Reply:
@reduz said that's ok to use avx as the minimum, and skylake apparently is the oldest win10 supported cpu architecture.

That said let me look at the march list.
Sandybridge is the first gpu that allowed AVX.

> ‘-march=sandybridge’
> Intel Sandy Bridge CPU with 64-bit extensions, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, POPCNT, AVX, AES and PCLMUL instruction set support.

Thoughts on switching to ` env.Prepend(CCFLAGS=["-march=skylake"])` and only `__AVX__` and SSE4.1?

Reply:
You're right it seems like we can only use SSE4.1.

Reply:
I wonder how much effort would it take to change the code to support multiple versions

Reply:
AFAIK passing `__SSE*__` or `__AVX*__` defines is bogus in any case. Those are defined by the compiler for the target platform, they're not meant to be passed manually. They should indeed be enabled using `-march`, etc., but that means we drop support for any architecture which doesn't support those extensions (i.e. Godot will crash on those platforms).

This of course means killing 32-bit support, but also support for some x86_64 platforms. And that won't work on aarch64 either.

I would suggest making an integration that doesn't attempt to use intrinsics initially, and look at those in a second step. This should be done properly (e.g. by @hpvb who knows a lot about this stuff).

Reply:
That's easy to do, thanks. We can do it as a two stage process.

Reply:
My goals for the next free time slot is to remove march and the defines for instrincs. I'll post the new performance numbers, but I don't want to retest Godot master.